cli/trampolines/common.h

```diff
@@ -0,0 +1,70 @@
+// Preprocessor annoyances
+#define CONCAT2(x,y)    x##y
+#define CONCAT(x,y)     CONCAT2(x, y)
+#define CNAMEADDR(name) CONCAT(CNAME(name),_addr)
+#define STR_(x)         #x
+#define STR(x)          STR_(x)
+#define I(x)            x
+
+// On macOS and 32-bit windows, we need to prepend underscores on symbols to match the C ABI
+#if defined(__APPLE__) || defined(_WIN32)
+#define UNDERSCORE(x) _##x
+#else
+#define UNDERSCORE(x)    x
+#endif
+
+// Windows requires some help with the linker when it comes to debuginfo/exporting
+#if defined(_WIN32) || defined(_WIN64)
+#define DEBUGINFO(name)     .def name; \
+                            .scl 2; \
+                            .type 32; \
+                            .endef
+#define EXPORT(name)        .section .drectve,"r"; \
+                            .ascii STR(-export:##I(name)); \
+                            .ascii " "; \
+                            .section .text
+#else
+#define DEBUGINFO(name)
+#define EXPORT(name)
+#endif
+
+// Windows 64-bit uses SEH
+#if defined(_WIN64)
+#define SEH_START1(name)    .seh_proc CNAME(name)
+#define SEH_START2()        .seh_endprologue
+#define SEH_END()           .seh_endproc
+#else
+#define SEH_START1(name)
+#define SEH_START2()
+#define SEH_END()
+#endif
+
+// If we're compiling with control-flow branch protection, mark the trampoline entry
+// points with `endbr{32,64}`, as appropriate on this arch
+#if defined(__CET__) && __CET__ & 1 != 0
+#if defined(__x86_64__)
+#define CET_START()     endbr64
+#else
+#define CET_START()     endbr32
+#endif
+#else
+#define CET_START()
+#endif
+
+// aarch64 on mac requires some special assembler syntax for both calculating memory
+// offsets and even just the assembler statement separator token
+#if defined(__aarch64__)
```

Comment:
why wouldn't this go in trampolines_aarch64.S, when it is only defined and used there?

Reply:
because it's nice to have the `.S` files be as free of preprocessor stuff as possible; it makes it much easier to, with your eyeball, compare the trampolines across platforms.